lib_gb/src/ppu/fifo/fetching_state.rs

```diff
@@ -0,0 +1,13 @@
+pub enum FetchingState{
+    FetchTileNumber,
+    FetchLowTile,
+    FetchHighTile,
+    Push,
+    Sleep
+}
+
+pub union FetchingStateData{
```

Comment:
This could become a struct with 3 Option fields, this will grant a better error handling and indication when a value is corrupted.

No need for unsafe code (even though its fun).